ankihub/gui/browser/browser.py

```diff
@@ -917,6 +928,64 @@ def _set_updated_today_tree_expanded_in_ui_config(expanded: bool):
     config.set_ui_config(ui_config)
 
 
+def _setup_smart_search_button() -> None:
+    dialog_manager_did_open_dialog.append(_on_dialog_manager_did_open_dialog)
+
+
+def _on_dialog_manager_did_open_dialog(
+    dialog_manager: aqt.DialogManager, dialog_name: str, dialog_instance: QWidget
+) -> None:
+    if dialog_name != "Browser":
+        return
+
+    add_smart_search_button_to_sidebar()
+
+
+def add_smart_search_button_to_sidebar():
+    if not config.public_config.get("ankihub_smart_search"):
+        return
+
+    smart_search_button = QToolButton()
+    smart_search_button.setIcon(sparkles_icon(theme_manager.night_mode))
+    smart_search_button.setIconSize(QSize(16, 16))
+    smart_search_button.setToolButtonStyle(Qt.ToolButtonStyle.ToolButtonIconOnly)
+    smart_search_button.setAutoRaise(True)
+    smart_search_button.setToolTip("AnkiHub Smart Search")
+
+    def on_deck_selected(dialog: SearchableSelectionDialog) -> None:
+        if not dialog.name:
+            return
+        ah_did = config.get_ah_did_by_name(dialog.name)
+        show_flashcard_selector(ah_did=ah_did, parent=browser)
```

Comment:
:wrench: We can fix the issue with the non-interactable login dialog using this:
```suggestion
        aqt.mw.taskman.run_on_main(
            lambda: show_flashcard_selector(ah_did=ah_did, parent=browser)
        )
```

The issue was caused because the modal selection dialog was still open when the login dialog got opened. Modal dialogs block the user from interacting with other dialogs.

When we use the `run_on_main` method of the task manager, we decouple showing the flashcard selector from the current function (`on_deck_selected`). This way the deck selection dialog closes before the error handlers opens the login dialog.

Reply:
@ddevdan If nothing else comes up, I think we can merge the PR after fixing the issue with the non-interactable login dialog

Reply:
Cool. I've just pushed the proposed change.